internal/controllers/synthesis/lifecycle.go

```diff
@@ -79,7 +79,7 @@ func (c *podLifecycleController) Reconcile(ctx context.Context, req ctrl.Request
 	syn.Name = comp.Spec.Synthesizer.Name
 	err = c.client.Get(ctx, client.ObjectKeyFromObject(syn), syn)
 	if err != nil {
-		return ctrl.Result{}, client.IgnoreNotFound(fmt.Errorf("getting synthesizer: %w", err))
+		return ctrl.Result{}, fmt.Errorf("getting synthesizer: %w", err)
```

Comment:
we shouldn't ever ignore a composition just because its synthesizer doesn't exist, assuming it has one configured. In this case a stale synth cache would cause synthesis not to happen until the next generation of the composition